### ISSUE TITLE
Panel title should use a heading tag

### ISSUE THREAD

**Issue by Stanton** [jadu/pulsar#999]
**Is your feature request related to a problem? Please describe.**
Panel titles are visually styled to represent a heading but are not marked up with a heading tag.

![Screenshot 2019-07-03 at 09 23 15](https://user-images.githubusercontent.com/18653/60575870-386dd500-9d74-11e9-88b2-c0bd0484f8c5.png)
https://pulsar.docs.jadu.net/components/panel 

This is required to ensure compliance with WCAG 2.1, put simply "Text that looks like a heading must be coded as a heading."

[1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
[2.4.6: Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html)

**Describe the solution you'd like**
Panel titles should be changed to use a heading tag. This may also require the default heading level to be changed if required.

**Comment by jamesjacobs** [comment]
Feels like we should elevate panels to region landmarks as they are pretty much guaranteed to contain information that’s important enough to want to be able to quickly SR navigate to. This would need `aria-labelledby="<heading-id>"` and either the wrapping `div` changing to a `section` or `role="region"` adding. (ideally `section`)
